OpenRA.Game/Map/MapCache.cs

```diff
@@ -131,7 +131,12 @@ public void LoadMap(string map, IReadOnlyPackage package, MapClassification clas
 						{
 							LastModifiedMap = uid;
 							if (oldMap != null)
-								mapUpdates.Add(oldMap, uid);
+							{
+								if (mapUpdates.ContainsKey(oldMap))
+									mapUpdates[oldMap] = uid;
+								else
+									mapUpdates.Add(oldMap, uid);
+							}
```

Comment:
`mapUpdates[oldMap] = uid;` alone is enough. It either adds or updates the value.

Reply:
or right